lib/Dialect/TTNN/Transforms/Optimizer.cpp

```diff
@@ -330,6 +341,18 @@ class TTNNOptimizer : public impl::TTNNOptimizerBase<TTNNOptimizer> {
                                          layoutAttr.getMemref().getShape())),
                 tensorMemoryLayoutAttr));
           }
+
+          // Set specific Conv2d Op configuration.
+          if (isa<ttnn::Conv2dOp>(op)) {
+            if (opConfigAnalysis.getResult().at(op).config) {
+              Attribute config = opConfigAnalysis.getResult().at(op).config;
+              if (isa<ttnn::Conv2dConfigAttr>(config)) {
+                ttnn::Conv2dOp conv2dOp = mlir::cast<ttnn::Conv2dOp>(op);
+                conv2dOp.setConv2dConfigAttr(
+                    mlir::cast<ttnn::Conv2dConfigAttr>(config));
+              }
+            }
+          }
```

Comment:
You have three nested ifs with no else
```suggestion
          if (auto conv2dOp = mlir::dyn_cast<ttnn::Conv2dOp>(op)) {
            if (auto conv2dConfig =
                    mlir::dyn_cast_if_present<ttnn::Conv2dConfigAttr>(
                        opConfigAnalysis.getResult().at(op).config)) {
              conv2dOp.setConv2dConfigAttr(conv2dConfig);
            }
          }
```